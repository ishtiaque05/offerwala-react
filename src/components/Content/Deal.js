import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';

import { 
  withStyles, 
  Card, 
  CardActionArea, 
  CardContent, 
  CardMedia, 
  Typography, 
  Hidden
} from '@material-ui/core';

import DefaultImage from '../../assets/images/default_deal.jpg';
import DesktopDealDetails from './DealDetails/DesktopDealDetails';
import MobileDealDetails from './DealDetails/MobileDealDetails';

const styles = theme => ({
  card: {
    width: theme.spacing.unit * 43.75,
    maxHeight: theme.spacing.unit * 100,
    margin: theme.spacing.unit * 1.5,
    background: '#FFF',
    boxShadow: '0 4px 6px #D4D4D4',
    [theme.breakpoints.down('xs')]: {
      marginLeft: 'auto',
      marginRight: 'auto', 
      width: '100%'
    }
  },
  flex: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  media: {
    height: 'auto'
  },
  link: {
    textDecoration: 'none'
  },
  shopName: {
    backgroundColor: '#4B4B4B',
    color: '#FFF',
    padding: theme.spacing.unit * 0.5,
    paddingLeft: theme.spacing.unit * 1.5,
    paddingRight: theme.spacing.unit * 1.5,
    borderRadius: theme.spacing.unit * 0.5
  },
  dealDetails: {
    margin: 0,
    padding: 0
  }
});

class Deal extends Component {
  state = {
    open: false
  };

  clickHandler = () => {
    this.setState({
      open: true
    });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes, deal } = this.props;
    console.log(deal)

    return (
      <Fragment>

        <Hidden smUp>
          <MobileDealDetails 
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          /> 
        </Hidden>
        <Hidden smDown>
          <DesktopDealDetails
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          />
        </Hidden>

        
        <Card onClick={this.clickHandler} className={classes.card}>
          <CardActionArea>
            <CardMedia
              style={{ padding: '16px', paddingBottom: '0px' }}
              component="img"
              alt="Contemplative Reptile"
              className={classes.media}
              image={
                deal.picture !== undefined ? deal.picture.url : DefaultImage
              }
              title={deal.title}
            />
            <CardContent>
              <Typography gutterBottom variant="h5" component="h2">
                {deal.title}
              </Typography>
              <div className={classes.flex}>
                <Typography className={classes.shopName}>
                  {deal.shop !== undefined ? deal.shop.title : ''}
                </Typography>
                <Typography>Ends: {deal.end_date}</Typography>
              </div>
            </CardContent>
          </CardActionArea>
        </Card>
      </Fragment>
    );
  }
}

Deal.propTypes = {
  classes: PropTypes.object.isRequired,
  deal: PropTypes.object.isRequired
};

export default withStyles(styles)(Deal);
