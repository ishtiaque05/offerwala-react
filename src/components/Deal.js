import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import {
  withStyles,
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
  Hidden
} from '@material-ui/core';

import DefaultImage from '../assets/images/default_deal.jpg';
import DesktopDealDetails from './DealDetails/DesktopDealDetails';
import MobileDealDetails from './DealDetails/MobileDealDetails';

const styles = theme => ({
  card: {
    width: '30%',
    maxHeight: theme.spacing.unit * 100,
    margin: theme.spacing.unit * 1.5,
    background: '#FFF',
    boxShadow: '0 4px 6px #D4D4D4',
    ['@media (min-width: 1200px)']: {
      width: '30%',
      marginLeft: '1%',
      marginRight: '1%'
    },
    ['@media (min-width: 1520px)']: {
      width: '23%',
      marginLeft: '1%',
      marginRight: '1%'
    },
    ['@media (min-width: 1921px)']: {
      width: '18%',
      marginLeft: '1%',
      marginRight: '1%'
    },
    [theme.breakpoints.down('sm')]: {
      marginLeft: '2%',
      marginRight: '2%',
      width: '46%'
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: 'auto',
      marginRight: 'auto',
      width: '100%'
    }
  },
  flex: {
    marginTop: '5px',
    [theme.breakpoints.down('sm')]: {
      display: 'flex',
      marginTop: 0,
      justifyContent: 'space-between'
    }
  },
  media: {
    height: 'auto'
  },
  link: {
    textDecoration: 'none'
  },
  shopName: {
    backgroundColor: '#4B4B4B',
    color: '#FFF',
    padding: theme.spacing.unit * 0.5,
    paddingLeft: theme.spacing.unit * 1.5,
    paddingRight: theme.spacing.unit * 1.5,
    borderRadius: theme.spacing.unit * 0.5,
    display: 'inline-block',
    marginBottom: '5px'
  },
  dealDetails: {
    margin: 0,
    padding: 0
  }
});

class Deal extends Component {
  state = {
    open: false
  };

  clickHandler = () => {
    this.setState({
      open: true
    });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes, deal } = this.props;

    return (
      <Fragment>
        <Hidden mdUp>
          <MobileDealDetails
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          />
        </Hidden>
        <Hidden smDown>
          <DesktopDealDetails
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          />
        </Hidden>

        <Card onClick={this.clickHandler} className={classes.card}>
          <CardActionArea>
            <CardMedia
              style={{ padding: '0px' }}
              component="img"
              alt="Contemplative Reptile"
              className={classes.media}
              image={
                deal.picture !== undefined ? deal.picture.url : DefaultImage
              }
              title={deal.title}
            />
            <CardContent>
              <Typography gutterBottom variant="h6">
                {deal.title}
              </Typography>
              <div>
                <Typography className={classes.shopName}>
                  {deal.shop !== undefined ? deal.shop.title : ''}
                </Typography>
                <Typography>Ends: {deal.end_date}</Typography>
              </div>
            </CardContent>
          </CardActionArea>
        </Card>
      </Fragment>
    );
  }
}

Deal.propTypes = {
  classes: PropTypes.object.isRequired,
  deal: PropTypes.object.isRequired
};

export default withStyles(styles)(Deal);
