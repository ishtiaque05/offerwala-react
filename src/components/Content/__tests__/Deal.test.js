import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Deal from 'components/Content/Deal';

describe('Deal', () => {
  let component;
  let testProps;

  beforeEach(() => {
    testProps = {
      deal: {
        imgUrl: 'default.jpg',
        title: ''
      }
    };
    component = shallow(<Deal {...testProps} />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Deal {...testProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
