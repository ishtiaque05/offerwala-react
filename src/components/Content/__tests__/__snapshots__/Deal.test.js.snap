// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal renders correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation1-6 MuiPaper-rounded-4 MuiCard-root-2 Deal-card-1"
>
  <button
    className="MuiButtonBase-root-33 MuiCardActionArea-root-30"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <img
      alt="Contemplative Reptile"
      className="MuiCardMedia-root-36 MuiCardMedia-media-37"
      height="140"
      src="default.jpg"
      title=""
    />
    <div
      className="MuiCardContent-root-38"
    >
      <h2
        className="MuiTypography-root-39 MuiTypography-h5-55 MuiTypography-gutterBottom-66"
      >
        
      </h2>
      <p
        className="MuiTypography-root-39 MuiTypography-body1-48"
      />
      <p
        className="MuiTypography-root-39 MuiTypography-body1-48"
      />
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-32"
    />
    <span
      className="MuiTouchRipple-root-74"
    />
  </button>
</div>
`;
